requests_ntlm/requests_ntlm.py

```diff
@@ -180,3 +178,32 @@ def _auth_type_from_header(header):
         return 'Negotiate'
 
     return None
+
+def _get_server_cert(request_url):
+    """
+    Get the certificate at the request_url and return it as a SHA256 hash. Will check the endpoint if it
+    is a https site and use the default port 443 is it isn't explicitly specified in the URL. Used to send
+    in with NTLMv2 authentication for Channel Binding Tokens
+
+    :param request_url: The request url in the format https://endpoint:port/path
+    :return: SHA256 hash of the DER encoded certificate at the request_url or None if not a HTTPS endpoint
+    """
+    host_pattern = re.compile('(?i)^https://?(?P<host>[0-9a-z-_.]+)(:(?P<port>\d+))?')
+    match = host_pattern.match(request_url)
+
+    if match:
+        host = match.group('host')
+        port = match.group('port')
+        if not port:
+            port = 443
+        else:
+            port = int(port)
+
+        s = socket()
+        c = ssl.wrap_socket(s)
+        c.connect((host, port))
+        server_certificate = c.getpeercert(True)
+        hash_object = hashlib.sha256(server_certificate)
+        return hash_object.hexdigest().upper()
```

Comment:
Ok, so if they use a different backend we should just ignore sending the cert and potentially get a failure if the server requires it or then use the hacky process I've got here in that case?